### ISSUE TITLE
ITestAction on assembly-level does not run for suites 

### ISSUE THREAD

**Issue by Sebazzz** [nunit/nunit#1806]
As discussed in [#1770](https://github.com/nunit/nunit/issues/1770#issuecomment-247627302), when an ITestAction is placed on the assembly level it does not run for test suites within the assembly. 

Please review an reproduction with [this gist](https://gist.github.com/Sebazzz/5c04b92f566bc8e9c6397b2f10d4f191). Compile by placing NUnit 3 console runner and NUnit 3 assembly into same directory and running 'run.bat'.

**Actual output:**
```
Before: C:\temp\nunittestaction\Test.dll
Before: TestActions.MyTestFixture.Test1
MyTestFixture.Test1
After: TestActions.MyTestFixture.Test1
Before: TestActions.MyTestFixture.Test2
MyTestFixture.Test2
After: TestActions.MyTestFixture.Test2
After: C:\temp\nunittestaction\Test.dll
```

**Expected output:**
```
Before: C:\temp\nunittestaction\Test.dll
Before: TestActions [note: since an namespace is an test suite too]
Before: TestActions.MyTestFixture
Before: TestActions.MyTestFixture.Test1
MyTestFixture.Test1
After: TestActions.MyTestFixture.Test1
Before: TestActions.MyTestFixture.Test2
MyTestFixture.Test2
After: TestActions.MyTestFixture.Test2
After: TestActions.MyTestFixture
After: TestActions
After: C:\temp\nunittestaction\Test.dll
```

Reproduces with NUnit 3.4.1 (current version).

**Comment by CharliePoole** [comment]
Additional information should go on the same issue, in this case #1770. I copied your comment there.

**Comment by Sebazzz** [comment]
My rationale for creating a new issue is because the referred issue is a question. Sorry.

**Comment by CharliePoole** [comment]
That's OK. The original poster was a little unclear what he wanted, so I marked it as a question. The issue has been hanging around for a while. I'll re-label it.
